frontend/src/components/Table/Table.test.tsx

```diff
@@ -0,0 +1,171 @@
+      data={(tableProps?.data ?? testData) as any}
+    />,
+    {
+      ...rest,
+    },
+  );
+  const tableRows = utils.container.querySelectorAll('.table .tbody .tr-wrapper');
+  return {
+    ...utils,
+    tableRows,
+  };
+};
+
+describe('Generic table component', () => {
+  beforeEach(() => {
+    setSort.mockClear();
+  });
+
+  it('matches snapshot', async () => {
+    const { asFragment } = setup({});
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('displays a message when no matching records found', async () => {
+    const { tableRows, getByText } = setup({ props: { data: [] } });
+
+    expect(tableRows.length).toBe(0);
+    expect(getByText('No rows to display')).toBeVisible();
+  });
+
+  it('by default the first page is the active page', async () => {
+    const { getByLabelText } = setup({ props: { pageSize: 5 } });
+
+    const page1Button = getByLabelText('Page 1 is your current page');
+    expect(page1Button.parentElement).toHaveClass('active');
+  });
+
+  it('it displays a maximum number of rows equal to the pagesize prop if manualPagination is false', async () => {
+    const { tableRows } = setup({ props: { pageSize: 5, manualPagination: false } });
+
+    expect(tableRows).toHaveLength(5);
+  });
+
+  it('it displays all rows if manualPagination is false', async () => {
+    const { tableRows } = setup({ props: { pageSize: 5, manualPagination: true } });
+
+    expect(tableRows).toHaveLength(6);
+  });
+
+  it('allows the current page to be changed', async () => {
+    const { getByLabelText, getByText } = setup({ props: { pageSize: 5 } });
+
+    const page2Button = getByLabelText('Page 2');
+    userEvent.click(page2Button);
+    expect(getByText('six')).toBeVisible();
+    expect(page2Button.parentElement).toHaveClass('active');
+  });
+
+  it('displays multiple pages when the total number of rows exceed the page size', async () => {
+    const { getByLabelText } = setup({ props: { pageSize: 5 } });
+
+    expect(getByLabelText('Page 2')).toBeVisible();
+  });
+
+  it('hides the paging control when hidePagination is set', async () => {
+    const { tableRows, queryByLabelText } = setup({ props: { pageSize: 5, hidePagination: true } });
+
+    const page1Button = queryByLabelText('Page 1 is your current page');
+    expect(page1Button).toBeNull();
+    expect(tableRows).toHaveLength(6);
+  });
+
+  it('does not change the page when manual pagination is set', async () => {
+    const onRequestData = jest.fn();
+    const { getByLabelText, tableRows } = setup({
+      props: { pageSize: 5, manualPagination: true, onRequestData },
+    });
+
+    const page2Button = getByLabelText('Page 2');
+    userEvent.click(page2Button);
+    expect(tableRows).toHaveLength(6); //pagination is handled externally, the table component will ignore the page size prop.
+    expect(onRequestData).toHaveBeenCalled();
+  });
+
+  it('can select only one row at a time when single select prop set', async () => {
```

Comment:
@Maheshbabu28 something like this was what I was hoping for when you fixed the single select pagination last sprint.